v3iofs/fs.py

```diff
@@ -37,49 +38,97 @@ class V3ioFS(AbstractFileSystem):
             endpoint=self._v3io_api,
             access_key=self._v3io_access_key,
-            transport_kind='requests',
+            transport_kind="requests",
         )
 
+    def _details(self, contents, **kwargs):
+        pathlist = []
+        for c in contents:
+            data = {}
+            data["name"] = c["name"].lstrip("/")
+            if c["size"] is not None:
+                data["type"] = "file"
+                data["size"] = c["size"]
+            else:
+                data["type"] = "directory"
+                data["size"] = 0
+            pathlist.append(data)
+        return pathlist
+
     def ls(self, path, detail=True, **kwargs):
         """Lists files & directories under path"""
-        container, path = split_container(path)
 
-        if not container:
+        container, path = split_container(path)
+        if container == "":
             return self._list_containers(detail)
+        else:
+            containers = self._list_containers(
+                detail=True
+                )
+            containers = [c["name"] for c in containers]
+            if container not in containers:
+                raise FileNotFoundError("Container not found!!")
 
         resp = self._client.get_container_contents(
-            container=container,
-            path=path,
-            get_all_attributes=True,
-            raise_for_status=[HTTPStatus.OK],
-        )
-
-        prefixes = resp.output.common_prefixes  # directories
-        fn = prefix_info if detail else prefix_path
-        dirs = [fn(container, p) for p in prefixes]
-
-        objects = resp.output.contents  # files
-        fn = object_info if detail else object_path
-        files = [fn(container, o) for o in objects]
-
-        return dirs + files
+                container=container,
+                path=path,
+                get_all_attributes=True,
+                raise_for_status="never",
+            )
+
+        # Try to fetch a list of directories, else return empty
+        if hasattr(resp.output, 'common_prefixes'):
+            prefixes = resp.output.common_prefixes
+            dirs = [prefix_info(container, p) for p in prefixes]
+        else:
+            dirs = []
+
+        # Try to fetch a list of files, else return empty
+        if hasattr(resp.output, 'contents'):
+            objects = resp.output.contents
+            files = [object_info(container, o) for o in objects]
+        else:
+            files = []
+
+        pathlist = dirs + files
+
+        if (not hasattr(resp.output, 'common_prefixes')) and (
+                not hasattr(resp.output, 'contents')):
+            dirname, _, filename = path.rpartition("/")
+            resp = self._client.get_container_contents(
+                container=container, path=dirname,
+            )
+            objects = resp.output.contents
+            tempfiles = [object_info(container, o) for o in objects]
+            fullpath = f"/{container}/{path}"
```

Comment:
We construct the original path after we destructed it. Looks like a code smell, maybe we can keep the original path in a variable.